openmc/deplete/integrator/cecm.py

```diff
@@ -1,133 +1,78 @@
+    r"""Deplete using the CE/CM algorithm.
 
-    Implements the second order CE/CM Predictor-Corrector algorithm [ref]_.
-    This algorithm is mathematically defined as:
+    Implements the second order `CE/CM predictor-corrector algorithm
+    <https://doi.org/10.13182/NSE14-92>`_.  This algorithm is mathematically
+    defined as:
 
     .. math::
         y' &= A(y, t) y(t)
 
         A_p &= A(y_n, t_n)
 
-        y_m &= \\text{expm}(A_p h/2) y_n
+        y_m &= \text{expm}(A_p h/2) y_n
 
         A_c &= A(y_m, t_n + h/2)
 
-        y_{n+1} &= \\text{expm}(A_c h) y_n
-
-    .. [ref]
-        Isotalo, Aarno. "Comparison of Neutronics-Depletion Coupling Schemes
-        for Burnup Calculations—Continued Study." Nuclear Science and
-        Engineering 180.3 (2015): 286-300.
+        y_{n+1} &= \text{expm}(A_c h) y_n
 
     Parameters
     ----------
-    operator : Operator
+    operator : openmc.deplete.TransportOperator
         The operator object to simulate on.
+    timesteps : iterable of float
+        Array of timesteps in units of [s]. Note that values are not cumulative.
+    power : float or iterable of float
+        Power of the reactor in [W]. A single value indicates that the power is
+        constant over all timesteps. An iterable indicates potentially different
+        power levels for each timestep. For a 2D problem, the power can be given
+        in [W/cm] as long as the "volume" assigned to a depletion material is
+        actually an area in [cm^2].
     print_out : bool, optional
         Whether or not to print out time.
-    """
-
-    # Save current directory
-    dir_home = os.getcwd()
 
-    # Move to folder
-    os.makedirs(operator.settings.output_dir, exist_ok=True)
-    os.chdir(operator.settings.output_dir)
+    """
+    if not isinstance(power, Iterable):
+        power = [power]*len(timesteps)
 
     # Generate initial conditions
-    vec = operator.initial_condition()
-
-    n_mats = len(vec)
-
-    t = 0.0
-
-    for i, dt in enumerate(operator.settings.dt_vec):
-        # Create vectors
-        x = [copy.deepcopy(vec)]
-        seeds = []
-        eigvls = []
-        rates_array = []
-
-        eigvl, rates, seed = operator.eval(x[0])
-
-        eigvls.append(eigvl)
-        seeds.append(seed)
-        rates_array.append(rates)
-
-        t_start = time.time()
+    with operator as vec:
+        chain = operator.chain
+        t = 0.0
+        for i, (dt, p) in enumerate(zip(timesteps, power)):
+            # Get beginning-of-timestep reaction rates
+            x = [copy.deepcopy(vec)]
+            op_results = [operator(x[0], p)]
 
-        chains = repeat(operator.chain, n_mats)
-        vecs = (x[0][i] for i in range(n_mats))
-        rates = (rates_array[0][i, :, :] for i in range(n_mats))
-        dts = repeat(dt/2, n_mats)
+            # Deplete for first half of timestep
+            x_middle = deplete(chain, x[0], op_results[0], dt/2, print_out)
```

Comment:
I'm not entirely thrilled with this format.  Take the CE/LI algorithm, for instance.  The second stage requires the matrix
expm(1/2 * F(beginning of step) + 1/2 * F(end of step)).  Technically, you can't interpolate the reaction rates before converting to matrix form, as the reaction rate to depletion matrix operator is slightly nonlinear (through power normalization).

This isn't a big deal for the time being.  If there's any interest in the new integrators, I can figure out a generic ``deplete`` function that will suit the needs for all of them.

Reply:
Sounds like a good plan. `deplete` is mostly an implementation detail and not meant to be called by a user directly, so if we want to add fancy integrators in the future, we can change the arguments at that time.